doc/development/plugins.rst

```diff
@@ -1,397 +1,318 @@
-inherits from Python's ``Sequence``, and represents an ordered set of unique wire labels.
-The ``labels`` attribute stores a tuple of the wire labels.
-Indexing a ``Wires`` instance with an integer will return the corresponding label.
-Indexing with a ``slice`` will return a ``Wires`` instance.
+1) Track and specify the exact configuration of the execution by only inspecting the ``ExecutionConfig`` object
+2) Dynamically configure the device over the course of a workflow.
 
-For example:
 
-.. code-block:: python
+**Workflow Configuration:**
 
-    from pennylane.wires import Wires
+Note that these properties are only applicable to devices that provided derivatives or VJPs. If your device
+does not provide derivatives, you can safely ignore these properties.
 
-    wires = Wires(['auxiliary', 0, 1])
-    print(wires.labels) # ('auxiliary', 0, 1)
-    print(wires[0]) # 'auxiliary'
-    print(wires[0:1]) # Wires(['auxiliary'])
+The workflow options are ``use_device_gradient``, ``use_device_jacobian_product``, and ``grad_on_execution``. 
+``use_device_gradient=True`` indicates that workflow should request derivatives from the device. 
+``grad_on_execution=True`` indicates a preference
+to use ``execute_and_compute_derivatives`` instead of ``execute`` followed by ``compute_derivatives``.  And finally
+``use_device_jacobian_product`` indicates a request to call ``compute_vjp`` instead of ``compute_derivatives``.  Note that 
+if ``use_device_jacobian_product`` is ``True``, this takes precedence over calculating the full jacobian.
 
-As shown in the section on :doc:`/introduction/circuits`, a device can be created with custom wire labels:
+>>> config = qml.devices.ExecutionConfig(gradient_method="adjoint")
+>>> processed_config = qml.device('default.qubit').preprocess(config)[1]
+>>> processed_config.use_device_jacobian_product
+True
+>>> processed_config.use_device_gradient
+True
+>>> processed_config.grad_on_execution
+True
 
-.. code-block:: python
+Execution
+---------
 
-    from pennylane import *
+For documentation on the expected result type output, please refer to :ref:`ReturnTypeSpec`.
 
-    dev = device('my.device', wires=['q11', 'q12', 'q21', 'q22'])
+The device API allows individual devices to calculate results in whatever way makes sense for
+the individual device. With this freedom over the implementation does come more responsibility
+to handle each stage in the process. 
 
-    @qnode(dev)
-    def circuit():
-       Gate1(wires='q22')
-       Gate2(wires=['q21','q11'])
-       Gate1(wires=['q21'])
-       return expval(Obs(wires='q11') @ Obs(wires='q12'))
+PennyLane does provide some helper functions to assist in executing
+circuits. Any ``StateMeasurement`` has ``process_state`` and ``process_density_matrix`` methods for
+classical post-processsing of a state vector or density matrix, and ``SampleMeasurement``'s implement
+both ``process_samples`` and ``process_counts``.  The ``pennylane.devices.qubit`` module also contains
```

Comment:
Can we also add examples to explain how this fits in the implementation of the `execute` method?

Reply:
better?